pkg/servicemesh/controller/memberroll/controller.go

```diff
@@ -78,31 +93,37 @@ func newMemberRollSharedInformer(restClient rest.Interface, namespace string, re
 }
 
 func (smmrc *serviceMeshMemberRollController) Start(stop chan struct{}) {
-	smmrc.lock.Lock()
-	defer smmrc.lock.Unlock()
+	if func() bool {
```

Comment:
Why an anonymous function instead of regular workflow?

Reply:
It's scoping the lock and ensuring it is released before the next section.  I was thinking about moving this into a separate function, it would likely improve the readability.

Reply:
I tidied this up and made it easier to read